eng/docgeneration/Generate-DocIndex.ps1

```diff
@@ -12,153 +27,91 @@ Write-Verbose "Initializing Default DocFx Site..."
-                }
-            } 
-            else 
-            {
-                if (($parent.IndexOf('azure-client-sdk-parent') -ne -1) -or ($parent.IndexOf('azure-data-sdk-parent') -ne -1))
-                {
-                    $clientArr += $artifactId
-                } 
-                else 
-                {
-                    $mgmtArr += $artifactId
-                }
-            }
-        }
+    else {
+        # Next page call
+        $resp = Invoke-RestMethod -Method Get -Uri "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=$langRegex&marker=$pageToken"
     }
-    # Only create this if there's something to create
-    #if (($clientArr.Count -gt 0) -or ($mgmtArr.Count -gt 0))
-    if ($clientArr.Count -gt 0)
-    {
-    New-Item -Path $YmlPath -Name "$($Dir.Name).md" -Force
-    Add-Content -Path "$($YmlPath)/toc.yml" -Value "- name: $($mappedDir)`r`n  href: $($Dir.Name).md"
-    # loop through the arrays and add the appropriate artifacts under the appropriate headings
-    if ($clientArr.Count -gt 0)
-    {
-        Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "# Client Libraries"
-        foreach($lib in $clientArr) 
-        {
-            Write-Host "Write $($lib) to $($Dir.Name).md"
-            Add-Content -Path "$($YmlPath)/$($Dir.Name).md" -Value "#### $lib"
+    # Storage returns some weired encoded string at the begining of response which needs to cutoff before use.
+    $rawConent = $resp.Substring(3)
+    # Convert to xml documents.
+    $xmlDoc = [xml]$rawConent
+    foreach ($elem in $xmlDoc.EnumerationResults.Blobs.BlobPrefix) {
+        # What service return like "dotnet/Azure.AI.Anomalydetector/", needs to fetch out "Azure.AI.Anomalydetector"
+        $aritifact = $elem.Name -replace $regex, '$1'
+        # Some languages need to convert the artifact name, e.g azure-data-appconfiguration -> @azure/data-appconfiguration
+        if ($packageNameRegex) {
+            $aritifact = $aritifact -replace $packageNameRegex, $packageNameReplacement
+        }
+        # Read the artifact package infomation from csv of Azure/azure-sdk/_data/release/latest repo.
+        $packageInfo = ($metadata | ? { $_.Package -Contains $aritifact})
+        # Ignore the one marked as Hide
+        $hidden = $packageInfo.Hide
+        if ($hidden -and $hidden.Trim() -eq "true") {
+            continue
+        }
+        $serviceName = $packageInfo.ServiceName
+        # If no service name retrieved, print out warning message, and put it into Other page.
+        if (!$serviceName) {
+            Write-Warning "Please double check and update the artifact correct service name to corresponding repo: https://github.com/Azure/azure-sdk/tree/master/_data/releases/latest."
+            Write-Warning "If the package is not relavant, please set Hide value to true."
+            $serviceName = "Other"
```

Comment:
I thought we were going to set this to the proper case of the service directory, or do we not have that information now?

Reply:
Right, we do not interate the sdk folder anymore. So we are not able to know the default one. Another way is to seek the sdk directory once we have the artifacts. But based on my testing, it seems not too many packages fall into Other page. We'd better to fix the csv file first.